llvm/unittests/Support/CommandLineTest.cpp

```diff
@@ -2206,4 +2206,25 @@ TEST(CommandLineTest, DefaultValue) {
   EXPECT_EQ(1, StrInitOption.getNumOccurrences());
 }
 
+TEST(CommandLineTest, UnknownSubcommand) {
+  cl::ResetCommandLineParser();
+
+  StackSubCommand SC1("foo", "Foo subcommand");
+  StackSubCommand SC2("bar", "Bar subcommand");
+  StackOption<bool> SC1Opt("f", cl::sub(SC1));
+  StackOption<bool> SC2Opt("b", cl::sub(SC2));
+  StackOption<bool> TopOpt("t");
```

Comment:
Perhaps worth adding a second top-level option to show that the right one is picked for the suggestion here too.

Reply:
Thanks. Done